index.js

```diff
@@ -78,6 +109,34 @@ class TrezorKeyring extends EventEmitter {
     return this.model;
   }
 
+  /**
+   * Gets the vendor, if known.
+   *
+   * @returns {"trezor" | "onekey"}
+   */
+  async getVendor() {
+    return this.vendor || (this.vendor = await this.fetchVendor());
```

Comment:
Currently `fetchVendor` will not be memoized as long as it returns null. In other words, if it returns null, then calling `getVendor` will call `fetchVendor` again. Is this the desired behavior?

Reply:
Yes, this is expected and will return the correct vendor name if the `TrezorConnect.getFeatures` method call is successful. If the call fails, then the function is expected to be called again.